docs/fl_introduction.rst

```diff
@@ -0,0 +1,68 @@
+.. _fl_introduction:
+
+###########################
+What is Federated Learning?
+###########################
+
+Federated Learning is a distributed learning paradigm where training occurs across multiple clients, each with their own local datasets.
+This enables the creation of common robust models without sharing sensitive local data, helping solve issues of data privacy and security.
+
+How does Federated Learning Work?
+=================================
+The federated learning (FL) aggregrator orchestrates the collaboration of multiple clients by first sending an initial model to the FL clients.
+The clients perform training on their local datasets, then send the model updates back to the FL aggregrator for aggregation to form a global model.
+This process forms a single round of federated learning and after a number of rounds, a robust global model can be developed.
+
+.. note::
+
+    In the diagrams below, the FL server has the role of the FL aggregrator. In the case of client-controlled workflows such as swarm learning,
+    FL clients can serve as FL aggregrators instead.
+
+.. image:: resources/fl_diagram.png
+    :height: 500px
+    :align: center
+
+FL Terms and Definitions
+========================
+
+- FL server: manages job lifecycle
+- FL aggregrator: orchestrates workflow, assigns tasks to clients, performs aggregation
+- FL client: executes tasks, performs local computation/learning with local dataset, submits result back to FL aggregrator
+
+.. image:: resources/controller_worker_flow.png
```

Comment:
agree to this.

In addition, do we want to touch these concepts here?
And right now in our code structure we seems to be combining/mixing a bit of "FL aggregator" and "FL server" together, not sure how should we approach that.

Reply:
Not sure, maybe it's better to just use "FL server" here instead of "FL aggregator" for simplicity in the overview, and just mention that in client-controlled case the clients can be the aggregator/controller?